Source/Core/DiscIO/Volume.h

```diff
@@ -22,17 +60,20 @@ class IVolume
 	virtual bool Read(u64 _Offset, u64 _Length, u8* _pBuffer, bool decrypt) const = 0;
 
 	virtual bool GetTitleID(u8*) const { return false; }
-	virtual std::unique_ptr<u8[]> GetTMD(u32 *_sz) const
-	{
+	virtual std::unique_ptr<u8[]> GetTMD(u32 *_sz) const {
 		*_sz = 0;
 		return std::unique_ptr<u8[]>();
 	}
 	virtual std::string GetUniqueID() const = 0;
 	virtual std::string GetMakerID() const = 0;
-	virtual int GetRevision() const { return 0; }
-	// TODO: eliminate?
-	virtual std::string GetName() const;
-	virtual std::vector<std::string> GetNames() const = 0;
+	virtual int GetRevision() const = 0;
+	virtual std::string GetName() const = 0;
+	virtual std::vector<std::pair<ELanguage, std::string>> GetNames() const = 0;
```

Comment:
vector of pair? Why not a map? Or do the same ```ELanguage``` values appear multiple times inside that list?